### PR TITLE
feature/Clear form after tag added [EOSF-73]

### PR DESCRIPTION
**Pull request by pattisdr**
## Ticket
https://openscience.atlassian.net/browse/EOSF-73

# Purpose

Intern request. Clears form after tag added.  Based off of patterns in @abought's commenting widget https://github.com/CenterForOpenScience/ember-osf/pull/61.

### CODE REVIEW

File: addon/mixins/taggable-mixin.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ export default Ember.Mixin.create({
             Ember.A(currentTags);
```
Comment (abought) [inline addon/mixins/taggable-mixin.js]: The return value is unused; what is this line doing?

Comment (pattisdr) [inline addon/mixins/taggable-mixin.js]: When I wrote this, I was having a hard time getting changedAttributes to update. I first did `file.get('tags').pushObject(tag)`, but it was saying that pushObject was not a function, so I changed it to Ember.A(file.get('tags')).pushObject(tag). But this results in changedAttributes being an empty object. So no PATCH request was sent.

The only way I could get changedAttributes to modify was to copy the current tags (not referencing original) and push new tag onto list, and then set tags equal to this list.  

Sam said that Henrique's work may fix this needing a roundabout way to modify the tags.

Comment (abought) [inline addon/mixins/taggable-mixin.js]: Right, but I think you've made two copies, and this second one appears unused. Does it work without this line?

(both `slice` and `Ember.Array` should return a new copy)

Comment (pattisdr) [inline addon/mixins/taggable-mixin.js]: It does not work without this line- it says that `currentTags.pushObject is not a function`.  I was using slice to make a copy and Ember.A to turn it into something I could do pushObject on.

Comment (pattisdr) [inline addon/mixins/taggable-mixin.js]: But you're right, I could simplify with `var currentTags = Ember.A(resource.get('tags'));`

Comment (abought) [inline addon/mixins/taggable-mixin.js]: Hmm. It appears that `Ember.A` really does work this way- weird, but interesting! Thanks.

Page suggesting it modifies the original in place: https://guides.emberjs.com/v2.3.0/configuring-ember/disabling-prototype-extensions/#toc_arrays

Page suggesting that it totally doesn't modify the original object:
http://emberjs.com/api/#method_A

Comment (pattisdr) [inline addon/mixins/taggable-mixin.js]: I think I still need the slice or something similar - I need to make a copy that doesn't reference the original in order for the changes to persist.

**Pull request merged**